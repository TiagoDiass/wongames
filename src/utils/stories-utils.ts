type Backgrounds = 'Dark' | 'Light' | 'Light Blue';

export const setStoryDefaultBackground = (background: Backgrounds) => ({
  backgrounds: {
    default: background
  }
});

type Layouts = 'fullscreen' | 'centered' | 'padded';

export const setStoryLayout = (storyLayout: Layouts) => ({
  layout: storyLayout
});
