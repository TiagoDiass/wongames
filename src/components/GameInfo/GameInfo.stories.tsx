import { ComponentMeta, ComponentStory } from '@storybook/react';
import { setStoryDefaultBackground } from 'utils/stories-utils';
import GameInfo from './GameInfo';
import GameInfoMock from './GameInfo.mock';

export default {
  title: 'Game/GameInfo',
  component: GameInfo,
  parameters: {
    ...setStoryDefaultBackground('Dark')
  },

  args: GameInfoMock
} as ComponentMeta<typeof GameInfo>;

export const Basic: ComponentStory<typeof GameInfo> = (args) => (
  <div style={{ maxWidth: '144rem', padding: '1.5rem', margin: 'auto' }}>
    <GameInfo {...args} />
  </div>
);
