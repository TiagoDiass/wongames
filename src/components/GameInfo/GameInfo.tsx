import * as S from './GameInfo.styles';
import { Heading, Button, Ribbon } from 'components';
import { AddShoppingCart, FavoriteBorder } from 'styled-icons/material-outlined';

export type GameInfoProps = {
  title: string;
  description: string;
  price: string;
};

/**
 * Component that will be used to display information about a game
 */
export default function GameInfo({ title, description, price }: GameInfoProps) {
  return (
    <S.Wrapper>
      <Heading color='black' lineBottom lineColor='primary'>
        {title}
      </Heading>

      <Ribbon color='secondary'>${price}</Ribbon>

      <S.Description>{description}</S.Description>

      <S.ButtonsWrapper>
        <Button icon={<AddShoppingCart />} size='large'>
          Add to cart
        </Button>

        <Button title='Add game to Wishlist' icon={<FavoriteBorder />} size='large' minimal>
          Wishlist
        </Button>
      </S.ButtonsWrapper>
    </S.Wrapper>
  );
}
