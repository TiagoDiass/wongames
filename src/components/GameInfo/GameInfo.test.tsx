import GameInfo from './GameInfo';
import { renderWithTheme } from 'utils/test-utils';
import { screen } from '@testing-library/react';

describe('Component: GameInfo', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(
      <GameInfo title='Game title' description='Game description' price='210.00' />
    );

    expect(screen.getByRole('heading', { name: 'Game title' })).toBeInTheDocument();
    expect(screen.getByText('Game description')).toBeInTheDocument();
    expect(screen.getByText('$210.00')).toBeInTheDocument();
    expect(screen.getByTitle('Add game to Wishlist')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Add to cart' })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
