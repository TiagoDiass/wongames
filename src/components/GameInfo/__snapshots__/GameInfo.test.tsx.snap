// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GameInfo should render correctly 1`] = `
.c3 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
  font-weight: 600;
  height: 3.6rem;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  right: -2rem;
  background-color: #3CD3C1;
}

.c3::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c3::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c0 {
  position: relative;
  background-color: #FAFAFA;
  padding: 2.4rem;
}

.c0 .c2 {
  right: -1rem;
}

.c0 .c2::before {
  border-right-width: 1rem;
}

.c4 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 0.8rem;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  border-radius: 0.4rem;
  outline: none;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c6:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c6:disabled:hover {
  -webkit-filter: none;
  filter: none;
}

.c6:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c6:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c6 svg {
  width: 1.6rem;
}

.c6 svg + span {
  margin-left: 0.8rem;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  border-radius: 0.4rem;
  outline: none;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: none;
  color: #F231A5;
}

.c8:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c8:disabled:hover {
  -webkit-filter: none;
  filter: none;
}

.c8:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c8:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c8 svg {
  width: 1.6rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c8:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c1 {
  color: #030517;
  font-weight: 600;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c1::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -1rem;
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c0 .c2 {
    top: 2.4rem;
    right: 2.4rem;
    font-size: 1.8rem;
  }

  .c0 .c2::before {
    border: none;
  }
}

@media (min-width:768px) {
  .c4 {
    max-width: 77rem;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-column-gap: 0.8rem;
    column-gap: 0.8rem;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    Game title
  </h2>
  <div
    class="c2 c3"
    color="secondary"
  >
    $
    210.00
  </div>
  <p
    class="c4"
  >
    Game description
  </p>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c8"
      title="Add game to Wishlist"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span>
        Wishlist
      </span>
    </button>
  </div>
</div>
`;
